File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int l = 0, r = nums.size()-1;
        while(l < r) {
            int x = nums[l] + nums[r];
            if(x > target) {
                r--;
            } else if(x < target) {
                l++;
            } else {
                return {l+1, r+1};
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> set = new HashSet<int>(nums);
        int ans = 0;
        foreach(int num in set) {
            if(!set.Contains(num-1)) {
                int streak = 1;
                while(set.Contains(num + streak)) {
                    streak++;
                }
                ans = Math.Max(ans, streak);
            }
        }
        return ans;
    }
}
